internal/graph/check.go

```diff
@@ -314,44 +336,64 @@ func exclusion(ctx context.Context, concurrencyLimit uint32, handlers ...CheckHa
 		wg.Done()
 	}()
 
+	var dbReads uint32
 	for i := 0; i < len(handlers); i++ {
 		select {
 		case baseResult := <-baseChan:
 			if baseResult.err != nil {
-				return &openfgapb.CheckResponse{Allowed: false}, baseResult.err
+				return &ResolveCheckResponse{
+					Allowed:            false,
+					ResolutionMetadata: &ResolutionMetadata{},
+				}, baseResult.err
 			}
 
+			dbReads += baseResult.resp.ResolutionMetadata.DatabaseReads
+
 			if !baseResult.resp.Allowed {
-				return &openfgapb.CheckResponse{Allowed: false}, nil
+				return &ResolveCheckResponse{
+					Allowed: false,
+					ResolutionMetadata: &ResolutionMetadata{
+						DatabaseReads: dbReads,
+					},
+				}, nil
 			}
+
 		case subResult := <-subChan:
 			if subResult.err != nil {
-				return &openfgapb.CheckResponse{Allowed: false}, subResult.err
+				return &ResolveCheckResponse{
+					Allowed:            false,
+					ResolutionMetadata: &ResolutionMetadata{},
```

Comment:
Similar to above, do we have inconsistent behaviour as we don't return dbRead that has been accumulated